autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +428,220 @@ model UserBlockCredit {
   blockId String?
   block   AgentBlock? @relation(fields: [blockId], references: [id])
 
+  // We need to be able to associate a credit transaction with an agent
+  executedAgentId      String?
+  executedAgentVersion Int?
+  executedAgent        Agent?  @relation(fields: [executedAgentId, executedAgentVersion], references: [id, version])
+
+  // We need to be able to associate a credit transaction with a store listing
+  storeListingId String?
+  storeListing   StoreListing? @relation(fields: [storeListingId], references: [id])
+
   amount Int
   type   UserBlockCreditType
 
-  isActive Boolean @default(true)
-  metadata Json?
+  isActive         Boolean         @default(true)
+  metadata         Json?
+  UserAccounting   UserAccounting? @relation(fields: [userAccountingId], references: [id])
+  userAccountingId String?
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+// This is the top level accounting table for a user
+model UserAccounting {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  transactions UserBlockCredit[]
+
+  usdBalance Float @default(0.0)
+
+  subscription   UserSubscription? @relation(fields: [subscriptionId], references: [id])
+  subscriptionId String?
+}
+
+// This table tracks a user's subscription to autogpt
+model UserSubscription {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isSubscribed          Boolean   @default(false)
+  subscriptionStartDate DateTime?
+
+  isCancelling        Boolean   @default(false)
+  subscriptionEndDate DateTime?
+
+  subscriptionPlanId String?
+  subscriptionPlan   SubsctiptionPlan? @relation(fields: [subscriptionPlanId], references: [id])
+
+  hasInitiatedChargeBackProcess Boolean          @default(false)
+  chargeBackProcessInitiatedAt  DateTime?
+  UserAccounting                UserAccounting[]
+
+  @@index([isSubscribed, isCancelling])
+}
+
+// This table tracks the different subscription plans available to a user
+model SubsctiptionPlan {
```

Comment:
@Torantulino please take a look at the accounting tables, as there are some assumptions baked in here about how it's going to work.

Thats this table and a few above it